Applications/Spire/Include/Spire/Ui/RegionListItem.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef SPIRE_REGION_LIST_ITEM_HPP
+#define SPIRE_REGION_LIST_ITEM_HPP
+#include <QWidget>
+#include "Nexus/Definitions/Region.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /**
+   * Represents a region item which can be a security, a market or a country
+   * in a RegionBox.
+   */
+  class RegionListItem : public QWidget {
+    public:
+
+      /**
+       * Constructs a RegionListItem.
+       * @param region The region which can be a security, a market or a country.
+       * @param parent The parent widget.
+       */
+      explicit RegionListItem(Nexus::Region region, QWidget* parent = nullptr);
+
+      /** Returns the region. */
+      const Nexus::Region& get_region() const;
+
+    private:
+      enum class Type {
```

Comment:
By default an ```enum class``` uses an ```int``` as its representation. When performance of reads/writes is insignificant, use a ```std::uint8_t``` to save on space.